pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,170 @@
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+
+import { Checkbox, Flex, FlexItem, Form, FormGroup, Popover, TextInput } from '@patternfly/react-core';
+import { show_modal_dialog, apply_modal_dialog } from "cockpit-components-dialog.jsx";
+import { HelpIcon } from '@patternfly/react-icons';
+
+import { has_errors } from "./dialog-utils.js";
+
+const _ = cockpit.gettext;
+
+function GroupCreateBody({ state, errors, change }) {
+    const {
+        name, id, user_specified_id_enabled,
+    } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup label={_("Name")}
+                       helperTextInvalid={errors && errors.name}
+                       validated={(errors && errors.name) ? "error" : "default"}
+                       fieldId="groups-create-name">
+                <TextInput id="groups-create-name"
+                           validated={(errors && errors.name) ? "error" : "default"}
+                           value={name} onChange={value => change("name", value)} />
+            </FormGroup>
+
+            <FormGroup label={_("ID")}
+                       hasNoPaddingTop
+                       helperTextInvalid={errors && errors.id}
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
+                    <Checkbox id="new-group-use-custom-id"
+                              label={_("Use specific ID")}
+                              isChecked={user_specified_id_enabled} onChange={checked => change("user_specified_id_enabled", checked)} />
+                    <FlexItem spacer={{ default: 'spacerLg' }}>
+                        <Popover bodyContent={_("When not specifying the ID the default is to use the smallest ID value greater than or equal to GID_MIN and greater than every other group.")}
+                                 showClose={false}>
+                            <HelpIcon />
+                        </Popover>
+                    </FlexItem>
+                </Flex>
+                {user_specified_id_enabled && <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />}
+            </FormGroup>
+        </Form>
+    );
+}
+
+function is_valid_char_name(c) {
+    return (c >= 'a' && c <= 'z') ||
+        (c >= 'A' && c <= 'Z') ||
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230109-123701-1958283d-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)